static/src/javascripts/projects/common/modules/commercial/epic-articles-viewed-opt-out.js

```diff
@@ -0,0 +1,169 @@
+
+const optOutEnabled = () => config.get('switches.showArticlesViewedOptOut');
+// Show the opt-out to 50% of the audience. We do not use `mvt % 2` here because then it would align
+// with the variants of the underlying A/B tests and distort the results
+const userIsInArticlesViewedOptOutTest = () => Number(getMvtValue()) < getMvtNumValues() / 2;
+
+type ArticlesViewedOptOutElements = {
+    checkbox: HTMLInputElement,
+    labelElement: HTMLElement,
+    optOutButton: HTMLElement,
+    optInButton: HTMLElement,
+    closeButton: HTMLElement,
+    buttons: HTMLElement,
+    header: HTMLElement,
+    body: HTMLElement,
+    note: HTMLElement,
+}
+
+const getElements = (element: HTMLElement): ?ArticlesViewedOptOutElements => {
+    const checkbox = element.querySelector('#epic-article-count__dialog-checkbox');
+
+    const labelElement = element.querySelector('.epic-article-count__prompt-label');
+    const optOutButton = element.querySelector('.epic-article-count__button-opt-out');
+    const optInButton = element.querySelector('.epic-article-count__button-opt-in');
+
+    const closeButton = element.querySelector('.epic-article-count__dialog-close');
+    const buttons = element.querySelector('.epic-article-count__buttons');
+    const header = element.querySelector('.epic-article-count__dialog-header');
+    const body = element.querySelector('.epic-article-count__dialog-body');
+    const note = element.querySelector('.epic-article-count__dialog-note');
+
+    if (
+        element &&
+        checkbox instanceof HTMLInputElement &&
+        labelElement &&
+        optOutButton &&
+        optInButton &&
+        closeButton &&
+        buttons &&
+        header &&
+        body &&
+        note
+    ) {
+        return {
+            element,
+            checkbox,
+            labelElement,
+            optOutButton,
+            optInButton,
+            closeButton,
+            buttons,
+            header,
+            body,
+            note
+        }
+    }
+};
+
+const setupHandlers = (elements: ArticlesViewedOptOutElements) => {
+
+    const hideDialog = () => {
+        // This is the hidden checkbox that we use to hide/unhide the dialog using css only
+        elements.checkbox.checked = false;
+    };
+
+    const onArticlesViewedClick = () => {
+        // Only need to send the tracking event here, as the dialog is displayed using only css
+        submitClickEvent({
+            component: {
+                componentType: 'ACQUISITIONS_OTHER',
+                id: 'articles-viewed-opt-out_open',
+            },
+        });
+    };
+
+    const onOptOutClick = () => {
+        submitClickEvent({
+            component: {
+                componentType: 'ACQUISITIONS_OTHER',
+                id: 'articles-viewed-opt-out_out',
+            },
+        });
+
+        addCookie(ARTICLES_VIEWED_OPT_OUT_COOKIE.name, new Date().getTime().toString(), ARTICLES_VIEWED_OPT_OUT_COOKIE.daysToLive);
```

Comment:
I think you should also zero-down the article counters at this point.

Reply:
👍